### PR TITLE
metrics and docs

### PR DESCRIPTION
**Pull request by casperdcl**
- metrics callback based on https://github.com/TomographicImaging/Hackathon-000-Stochastic-QualityMetrics
- improved local run docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,24 +28,30 @@ You can also find some example notebooks here which should help you with your de
...
-The organisers will effectively execute:
+The organisers will execute (after downloading https://petric.tomography.stfc.ac.uk/data/ to `/path/to/data`):
 
 ```sh
-docker run --rm -it -v data:/mnt/share/petric:ro ghcr.io/synerbi/sirf:edge-gpu
-# or ideally ghcr.io/synerbi/sirf:latest-gpu after the next SIRF release!
-python
+docker run --rm -it -v /path/to/data:/mnt/share/petric:ro -v .:/workdir -w /workdir synerbi/sirf:edge-gpu /bin/bash
+# ... or ideally synerbi/sirf:latest-gpu after the next SIRF release!
+pip install git+https://github.com/TomographicImaging/Hackathon-000-Stochastic-QualityMetrics
```
Comment (KrisThielemans) [inline README.md:36]: Why not add this to our docker image? But this is fine of course

File: petric.py [hunk 2]
```diff
@@ -66,20 +69,30 @@ def __call__(self, algo: Algorithm):
...
+    def __init__(self, seconds=300, outdir=OUTDIR, transverse_slice=None, coronal_slice=None, ground_truth=None,
+                 verbose=1):
         super().__init__(verbose)
         self.callbacks = [
             cbks.ProgressCallback(),
             SaveIters(outdir=outdir),
-            TensorBoard(logdir=outdir, transverse_slice=transverse_slice, coronal_slice=coronal_slice)]
+            (tb_cbk := TensorBoard(logdir=outdir, transverse_slice=transverse_slice, coronal_slice=coronal_slice))]
+
+        if ground_truth:
```
Comment (KrisThielemans) [inline petric.py:88]: I prefer to rename ground_truth to converged_image

File: petric.py [hunk 3]
```diff
@@ -66,20 +69,30 @@ def __call__(self, algo: Algorithm):
...
+            roi_image_dict = {f'S{i}': STIR.ImageData(f'S{i}.hv') for i in range(1, 8)}
+            self.callbacks.append(
+                ImageQualityCallback(
```
Comment (KrisThielemans) [inline petric.py:91]: Add a comment that this is for testing only, and that these are not part of the metrics to be used for the challenge

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
Looks good, but I haven't checked in a lot of detail, but go ahead. 

I just have some comments on naming etc, to avoid confusing participants

File: README.md (hunk 1 above)
Comment (casperdcl) [reply README.md:36]: I think it'll change a lot right now, and eventually be baked in to CIL itself, so wouldn't add it to the SIRF image for now.

File: petric.py (hunk 2 above)
Comment (casperdcl) [reply petric.py:88]: or `reference`?

**Comment by KrisThielemans** [comment]
Looks like I cannot review again via the app. But it's good to go!

**Pull request merged**